packages/recurring-payments-job/src/__tests__/recurring-payments-processor.spec.js

```diff
@@ -588,6 +605,112 @@ describe('recurring-payments-processor', () => {
     expect(debugLogger).toHaveBeenCalledWith(expect.any(String), error)
   })
 
+  // --- //
+
+  it('should log errors from await salesApi.processRPResult', async () => {
+    salesApi.getDueRecurringPayments.mockResolvedValueOnce([getMockDueRecurringPayment()])
+    salesApi.createTransaction.mockResolvedValueOnce({ id: 'trans-1', cost: 30 })
+
+    const payment = makeSendPaymentResponse()
+    sendPayment.mockResolvedValueOnce(payment)
+
+    getPaymentStatus.mockResolvedValueOnce(getPaymentStatusSuccess())
+    salesApi.processRPResult.mockRejectedValueOnce(new Error('boom'))
+
+    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
+
+    await execute()
+
+    expect(errorSpy).toHaveBeenCalledWith('Failed to process Recurring Payment for trans-1', expect.any(Error))
```

Comment:
Would be good to make sure the same error you set up earlier is the one being called here. Otherwise this could get a false pass from throwing a different error.